### PR TITLE
docs: multi pub-sub arch in barch service

### PR DESCRIPTION
**Pull request by Ubisoft-potato**
This PR proposes a new architecture for multiple GCP pub-sub.

### CODE REVIEW

File: docs/rfcs/batch-service-pubsub-improvement.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+- **Configuration**: The configuration for the subscriber, which includes the topic and the subscription name.
+- **Subscriber**: The subscriber that will pull the messages from the PubSub.
+- **Processor**: The processor that will process the messages from the PubSub.
+
+We will create a multi pub-sub Golang struct to handle the configuration, subscriber, and processor. The multi pub-sub
+will have the following methods:
+
+- **NewMultiPubSub**: The constructor for the multi pub-sub.
+- **Start**: The method to start the multi pub-sub.
+- **Stop**: The method to stop the multi pub-sub.
```
Comment (cre8ivejp) [inline docs/rfcs/batch-service-pubsub-improvement.md:27]: When will the stop be called? When is the batch server's context done?

File: docs/rfcs/batch-service-pubsub-improvement.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+```
+
+The Subscriber struct have a `Run` function to start the subscriber. The `Run` function will pull the messages from the
+PubSub and call the processor function to process the messages.
+
+```go
+package subscriber
+
+func (s Subscriber) Run(ctx context.Context) {
```
Comment (cre8ivejp) [inline docs/rfcs/batch-service-pubsub-improvement.md:82]: Will the `Start` be called the `Run`?

https://github.com/bucketeer-io/bucketeer/pull/877/files#diff-66a757d8cbd783a760f5b06910a941074eb3e2e1fbef97962aa34deb00b49fb0R145

File: docs/rfcs/batch-service-pubsub-improvement.md (hunk 1 above)
Comment (Ubisoft-potato) [reply docs/rfcs/batch-service-pubsub-improvement.md:27]: Yes, the `Stop` function will be called when the server context is done or receives an exit signal.

File: docs/rfcs/batch-service-pubsub-improvement.md (hunk 2 above)
Comment (Ubisoft-potato) [reply docs/rfcs/batch-service-pubsub-improvement.md:82]: Correct, we will call `Run` function of each subscriber in `Start`.

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**